opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -287,8 +293,15 @@ namespace Opm
         } else {
             from = WellProducerCMode2String(ws.production_cmode);
         }
+        // Only keep first and last control until iterationIdx >= nupcol (These are used for output)
+        // After iteration >= nupcol we log all switches to check if the well controls oscillates
+        const int episodeIdx = simulator.episodeIndex();
+        const int iterationIdx = simulator.model().newtonMethod().numIterations();
+        const int nupcol = schedule[episodeIdx].nupcol();
+        if (iterationIdx < nupcol && this->well_control_log_.size() > 2) {
```

Comment:
From the code, during the local solve, we only update the `well_controls_log_`  after the local solve getting converged. Do we need to do this `erase` procedure during local solve here, will it conflict with the `erase` from the outer iteration?  Let me know if it happens that I read the code wrong.